In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('clima.csv')

## 1. Mostre as 10 primeiras temperaturas do dataset para MinTemp.

In [4]:
minTemp10 = df['MinTemp']
minTemp10[0:10]

0    13.4
1     7.4
2    12.9
3     9.2
4    17.5
5    14.6
6    14.3
7     7.7
8     9.7
9    13.1
Name: MinTemp, dtype: float64

In [5]:
df['MinTemp'].head(10)

0    13.4
1     7.4
2    12.9
3     9.2
4    17.5
5    14.6
6    14.3
7     7.7
8     9.7
9    13.1
Name: MinTemp, dtype: float64

## 2 - Que tipo de informação essa coluna representa?

In [6]:
df['MinTemp'].dtype

dtype('float64')

## 3 - Calcule as seguintes medidas estatísticas de MinTemp:
Média, 
Mediana, 
Desvio Padrão, 
Mínimo, 
Máximo, 

In [7]:
# Média
df['MinTemp'].mean()

np.float64(12.194034380968919)

In [8]:
# Mediana 
df['MinTemp'].median()

np.float64(12.0)

In [10]:
# Desvio Padrão
df['MinTemp'].std()

np.float64(6.398494975663227)

In [11]:
# Minimo
df['MinTemp'].min()

np.float64(-8.5)

In [12]:
# Máximo
df['MinTemp'].max()

np.float64(33.9)

## 4 - Agora verifique quais valores de temperatura mínima aparecem com mais frequência (12).
## Existe alguma temperatura que aparece muitas vezes 

In [13]:
freq = df['MinTemp'].value_counts()
print('Temperatura mais frequente em MinTemp: ', freq.idxmax())

Temperatura mais frequente em MinTemp:  11.0


# 1.Em média, qual é a temperatura mínima registrada ?
### A menor temperatura registrada foi: 12.19

# 2. As temperaturas mínimas variam muito ou pouco ? (Dica: observe o desvio padrão)
### As temperaturas mínimas variam muito. Os dados são bem dispersos e não se mantêm próximos da média de $12,19°C$.


# 3.Qual foi a menor temperatura registrada no dataset ?
### A menor temperatura foi: -8.5

# 4.Qual foi a maior temperatura mínima registrada?
### A maior temperatura foi: 33.9

# 5. A maioria das temperaturas mínimas está próxima da média ou muito distante dela
### A maioria das temperaturas mínimas está muito distante da média. Isso se confirma pelo alto valor do desvio padrão ($6,40$), que indica uma grande dispersão dos dados em torno do valor central ($12,19$). Na prática, os valores variam significativamente de um dia para o outro.


In [17]:
## 2. As temperaturas mínimas variam muito ou pouco ? (Dica: observe o desvio padrão)
media = df['MinTemp'].mean()
dp = df['MinTemp'].std()
cv = (dp/media) * 100
print(cv)

52.47233832347792
